### PR TITLE
Bump pnpm/actions-setup version to 4.0.0

### PR DESCRIPTION
**Pull request by zaporter-work**
https://github.com/pnpm/pnpm/issues/6424

Causing
```
Running self-installer...
   WARN  GET https://registry.npmjs.org/pnpm error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
   WARN  GET https://registry.npmjs.org/pnpm error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
   ERR_PNPM_META_FETCH_FAIL  GET https://registry.npmjs.org/pnpm: Value of "this" must be of type URLSearchParams
Error: Something went wrong, self-installer exits with code 1
```
in app

More info in private slack thread https://viaminc.slack.com/archives/C02KFSY4H89/p1720023335832159

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   pull_request:
     branches:
       - main
+      - zp-bump
```
Comment (zaporter-work) [inline .github/workflows/ci.yaml:7]: will remove before merge

**Review by ethanlook — APPROVED** [review]

**Comment by lord-Sid** [comment]
After this bump, our action started failing ->

Can you please look into it. Thanks

image

**Comment by zaporter-work** [comment]
@lord-Sid Are you able to share your github actions / github repo that this is failing on?

**Comment by lord-Sid** [comment]
@zaporter-work This is my test.yml actions file contents ->

name: Run Tests and Linters
on:
  workflow_dispatch:
  pull_request:

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: 📥 Checkout code
        uses: actions/checkout@v3

      - name: 🏭 Setup pnpm
        uses: pnpm/action-setup@v2
        with:
          version: 8.6.9

      - name: 📦 Install dependencies
        run: pnpm install --frozen-lockfile

      - name: 💠 Generate Prisma Client
        run: pnpm run prisma-generate --generator client

      - name: 💅 Run linter
        run: pnpm lint

      - name: 🧪 Run tests with coverage
        run: pnpm test:cov --coverageReporters json-summary

      - name: 💬 Coverage Comment
        uses: MishaKav/jest-coverage-comment@main

**Comment by zaporter-work** [comment]
@lord-Sid I suspect your issue was not caused by us (because you don't seem to be using viamrobotics/js-config (unless you are doing it elsewhere)), instead, I suspect we both ran into the same issue.


I suspect you need to change your setup-pnpm to be::
```
  - name: 🏭 Setup pnpm
    uses: pnpm/action-setup@23657c8550e9be4f7ee7e9d7fe8adda6d703bb4e # v4.0.0
    with:
      version: 8.6.9
```

**Comment by lord-Sid** [comment]
@zaporter-work Thanks ! It was indeed same issue. Bump up to v4 fixed it :)

**Pull request merged**